scripts/core/ui/components/SelectUser.tsx

```diff
@@ -1,120 +1,139 @@
-                        fetchedUsers: res._items,
-                        loading: false,
-                    });
-                }
-            });
+        this.abortController = null;
     }
 
     componentDidMount() {
         this._mounted = true;
 
-        this.queryUsers();
+        if (this.props.selectedUserId != null) {
+            dataApi.findOne<IUser>('users', this.props.selectedUserId).then((selectedUser) => {
+                this.setState({selectedUser});
+            });
+        }
     }
 
     componentWillUnmount() {
         this._mounted = false;
     }
 
+    componentDidUpdate(prevProps: IPropsSelectUser) {
+        if (prevProps.selectedUserId !== this.props.selectedUserId && this.props.selectedUserId != null) {
+            dataApi.findOne<IUser>('users', this.props.selectedUserId).then((selectedUser) => {
+                this.setState({selectedUser});
+            });
+        }
+    }
+
     render() {
-        const keyedUsers = keyBy(this.state.fetchedUsers, (user) => user._id);
+        if (this.state.selectedUser === 'loading') {
+            return null;
+        }
 
         return (
-            <Select2
-                autoFocus={this.props.autoFocus ?? true}
-                disabled={this.props.disabled}
-                placeholder={(
-                    <ListItem fullWidth noBackground noShadow>
-                        <ListItemColumn ellipsisAndGrow>
-                            <ListItemRow>{gettext('Select a user')}</ListItemRow>
-                        </ListItemColumn>
-                    </ListItem>
-                )}
-                value={this.props.selectedUserId == null ? undefined : this.props.selectedUserId}
-                items={keyedUsers}
-                getItemValue={(user) => user._id}
-                onSelect={(value) => {
-                    this.props.onSelect(keyedUsers[value]);
+            <SelectWithTemplate
+                getItems={(searchString) => {
+                    this.abortController?.abort();
+                    this.abortController = new AbortController();
+
+                    const query = JSON.stringify(
+                        searchString != null && searchString.length > 0
+                            ? {
+                                $or: [
+                                    {
+                                        display_name: {
+                                            $regex: searchString,
+                                            $options: '-i',
+                                        },
+                                    },
+                                    {
+                                        username: {
+                                            $regex: searchString,
+                                            $options: '-i',
+                                        },
+                                    },
+                                ],
+                            }
```

Comment:
seems like we have multiple versions of doing this query, would be good to put this in a helper and make it consistent everywhere, we're searching also `email/first_name/last_name` in other places